setup.py

```
@@ -104,7 +104,7 @@ def run(self):
           "docker",
           "pymongo",
           "gdown",
-          "htcondor"
+#          "htcondor"
```

We need to work out a way to make this optional? Can a dependency be made to fail silently?

---

good question about failing silently. I'm not sure about that however we can have optional dependencies. The problem however is that it's easy to optionally have an extra dependency but not sure how to optionally have less. I'll have to have a look

---

In the branch of the fork, I've modified [setup.py](https://github.com/kh296/ganga/blob/9df995563543e7dc9da3f5ff8e61268fbfda2d1f/setup.py#L96) to make inclusion of `htcondor` in the list `install_requires` conditional on platform.  I'm not sure if this can be propagated to pypi, but it at least allows installation from github.

I've also changed the way in which a failed import of `htcondor` is handled in [ganga/GangaCore/Lib/Condor/Condor.py](https://github.com/kh296/ganga/blob/9df995563543e7dc9da3f5ff8e61268fbfda2d1f/ganga/GangaCore/Lib/Condor/Condor.py#L21).  (The handling based on warnings didn't seem very robust.)

I've tested that, after these changes, the simple job splitting example:

```
j = Job()
j.application.args = []
j.splitter=ArgSplitter(args=[['Hello 1'], ['Hello 2']])
j.submit()
```

runs successfully for me on the Local backend for both MacOS and Linux, and runs on the Condor backend for Linux.

---

We already make the LHCb conditional so we can do the same for htcondor. Though I would argue the default should include condor as its in GangaCore.

---

The condition added in [setup.py](https://github.com/kh296/ganga/blob/9df995563543e7dc9da3f5ff8e61268fbfda2d1f/setup.py#L96) includes `htcondor` in the dependencies except for MacOS (for which `htcondor` isn't available on pypi):

```
if sys.platform != 'darwin':
      install_requires.append("htcondor")
```

There's probably a better way of doing this, and any solution that allows for installing ganga without requiring `htcondor` would work for me :)

---

I agree with @mesmith75 that the default should be to include it since it's in GangaCore and I'm happy with the suggested fix to `setup.py` as long as it works installing from PyPI